src/openslide-decode-tifflike.c

```diff
@@ -1134,3 +1134,46 @@ bool _openslide_tifflike_init_properties_and_hash(openslide_t *osr,
 
   return true;
 }
+
+bool _openslide_tifflike_icc_profile_size(openslide_t *osr G_GNUC_UNUSED,
+                                          struct _openslide_tifflike *tl,
+                                          int32_t property_dir,
+                                          int64_t *len) {
+  struct tiff_item *item = get_item(tl, property_dir, TIFFTAG_ICCPROFILE);
+  if (item) {
+    *len = item->count;
+  } else {
+    *len = 0;
+  }
+
+  return false;
+}
+
+bool _openslide_tifflike_icc_profile_read(openslide_t *osr G_GNUC_UNUSED,
```

Comment:
I think let's drop this for now.  The only driver that can't use libtiff at runtime is NDPI, and it extracts everything it needs from the slide at open time.  If NDPI ends up needing ICC support, we should probably extend tifflike to return an offset/length for the value (and then have the driver cache the result), rather than requiring NDPI to reparse all the TIFF directories at runtime to call this function.

Reply:
OK, I copied the code into the PR as a reminder.